pkg/proc/variables.go

```diff
@@ -1233,12 +1233,11 @@ func (v *Variable) maybeDereference() *Variable {
 			return &v.Children[0]
 		}
 		ptrval, err := readUintRaw(v.mem, v.Addr, t.ByteSize)
-		r := v.newVariable("", ptrval, t.Type, DereferenceMemory(v.mem))
 		if err != nil {
-			r.Unreadable = err
+			v.Unreadable = err
```

Comment:
The change to this function should be reverted, it should still return a new, dereferenced, variable, even if there is a read error. Otherwise pointers that contain a bad address will end up with themselves as the first child.